aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -1174,6 +1174,8 @@ public void publishData(byte[] data, String topic, AWSIotMqttQos qos,
                             AWSIotMqttMessageDeliveryCallback.MessageDeliveryStatus.Fail,
                             userData);
                 }
+            } else {
+            	throw new AmazonClientException("Client is reconnecting and publishing from offline queue is disabled.");
```

Comment:
This covers the case where a message is dropped where offlinePublishQueueEnabled is false.

What do you think about line 1171? It seems that putMessageInQueue, could return true and drop a message if fullQueueKeepsOldest is false.  It seems it will drop the oldest message silently and return true.  Perhaps this does not affect our use case.